### DISCUSSION TITLE
How can I use ExecuTorch to deploy a model to a MicroController,such as Infineon TC3xxx ?

### DISCUSSION THREAD

**Discussion by AlexLuya** [pytorch/executorch#8217] [Q&A] [answered]
"ExecuTorch is an end-to-end solution for enabling on-device inference capabilities across mobile and edge devices including wearables, **embedded devices** and **microcontrollers**"

Hello,above expression presents in [ExecuTorch doc:](https://pytorch.org/executorch/stable/intro-overview.html)

I want to know:

what types of MicroController(mainly bare metals) got supported already or will get supported?

If wanting to deploy to Infineon TC3xxx microcontroller,is it possible?If yes,any suggestion about how to do it?

**Comment by iseeyuan** [comment]
The short answer is Yes. ExecuTorch is targeted at portability. The runtime code is lean and can be built and run in MicroControllers. We've successfully built and run ExecuTorch runtime in ARM microcontrollers (like M55/U55) and Cadence Tensilica DSPs. You may refer to arm and cadence folders under executorch/backends and executorch/examples for reference.

**Comment by mergennachin** [comment, marked as answer]
https://pytorch.org/executorch/main/build-run-xtensa.html

https://pytorch.org/executorch/main/executorch-arm-delegate-tutorial.html
